.github/workflows/ci.yml

```diff
@@ -117,9 +117,13 @@ jobs:
       - name: "Setup environment"
         run: "pip install ruff==0.5.0"
       - name: "Linting: ruff check"
-        run: "ruff check ."
+        run: "ruff check . --exclude python_sdk"
       - name: "Linting: ruff format"
-        run: "ruff format --check --diff ."
+        run: "ruff format --check --diff --exclude python_sdk ."
+      - name: "Linting: ruff check [SDK]"
+        run: "ruff check python_sdk --config python_sdk/pyproject.toml"
+      - name: "Linting: ruff format [SDK]"
+        run: "ruff format --check --diff --config python_sdk/pyproject.toml python_sdk"
```

Comment:
when we run `ruff check .` we don't use the config for mypy defined in `python_sdk/pyproject.toml` which led to some false positive failures. Everything works properly when we run `invoke sdk.lint`

Since we're planning to move the code for the SDK anyway I didn't want to change too many things as part of this PR but I think we should refactor `python-lint` in GHA to use the proper configs for each directory